### DISCUSSION TITLE
Change _Backend.delete_entry to accept UUIDs instead of Entries?

### DISCUSSION THREAD

**Discussion by shilorigins** [pcdshub/superscore#29] [General]
`_Backend.delete_entry` requires an `Entry` as input, which raises a question: what if the entry's UUID exists in the backend, but the entry itself has different data?  Should this raise an error because the arg entry "doesn't exist"?  If not, we should really change the signature to accept a UUID to remove this ambiguity.

**Comment by tangkong** [comment]
I think it's reasonable to raise an error if the contents of the Entry are quite different, but if that happens that suggests something more insidious is going on.  If the uuid and data don't match that could mean: 
* Someone has modified the entry locally and not synced the changes with the backend
* The backend has internal consistency issues

There are probably other issues but in either case I think it's probably prudent to warn the user if either of these cases is true

**Reply by shilorigins** [reply]
Sounds good to me.
